### PR TITLE
Airport-Challenge - 1

### PR DESCRIPTION
**Pull request by benja2208**

### CODE REVIEW

**Comment by zdajani** [comment]
really great, just needed to create doubles of the plane in airport rspec for one or two of the tests, which was done during the one-to-one.

File: lib/airport.rb [hunk 1]
```diff
@@ -1,2 +1,41 @@
...
+
+	def initialize ( capacity = DEFAULT_CAPACITY )
+		@planes = []
+		@capacity = capacity
+	end 
+
+	def release plane
+		raise 'Cannot take off while weather is stormy' if stormy? == true
+		raise 'plane is already flying' if plane.flying? == true
+		@planes.pop
```
Comment (silvabox) [inline lib/airport.rb]: !! `@planes.pop` will remove the last element from the array - *this is not necessarily the plane you've instructed to take off*.

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -1,48 +1,82 @@
...
-    # The weather must be random and only have two states "sunny" or "stormy".
-    # Try and take off a plane, but if the weather is stormy,
-    # the plane can not take off and must remain in the airport.
-    #
-    # This will require stubbing to stop the random return of the weather.
-    # If the airport has a weather condition of stormy,
-    # the plane can not land, and must not be in the airport
+      it 'test if Airport is full' do
+        allow(subject).to receive(:stormy?).and_return(false)
+        subject.capacity.times{subject.receive Plane.new}
```
Comment (silvabox) [inline spec/airport_spec.rb]: You should be using doubles here - not `Plane`

File: spec/airport_spec.rb [hunk 3]
```diff
@@ -1,48 +1,82 @@
...
-# No more planes can be added to the airport, if it's full.
-# It is up to you how many planes can land in the airport
-# and how that is implemented.
-#
-# If the airport is full then no planes can land
-
 describe Airport do
+  let(:landed_plane) {double(:landed_plane, :flying? => false)}
+  let(:flying_plane) {double(:flying_plane,:flying? => true)}
+  let(:landed_plane_stormy) {double(:landed_plane_stormy,:flying? => false, :stormy? => true)}
```
Comment (silvabox) [inline spec/airport_spec.rb]: Are these used?  Delete unused code.  Note also you have stubbed `stormy?` on a plane double.

**Comment by silvabox** [comment]
Hi Benjamina, thanks for submitting your airport challenge.  You need to fix your indentation to 2 spaces instead of tab.  It's a nice clean solution but there is on major issue - one with `@planes.pop`.  This demonstrates that you need more tests...

Please study the common issues here: https://github.com/makersacademy/airport_challenge/wiki/Common-issues and see how these relate to your own code.

:checkered_flag:

**Pull request closed without merging**